JavaScript/JavaScriptSDK/Context/Session.ts

```diff
@@ -87,32 +87,22 @@ module Microsoft.ApplicationInsights.Context {
             }
         }
 
+        public backup() {
+            this.setStorage(this.automaticSession.id, this.automaticSession.acquisitionDate, this.automaticSession.renewalDate);
+        }
+
         private initializeAutomaticSession() {
             var cookie = Util.getCookie('ai_session');
             if (cookie && typeof cookie.split === "function") {
-                var params = cookie.split("|");
-                if (params.length > 0) {
-                    this.automaticSession.id = params[0];
-                }
-
-                try {
-                    if (params.length > 1) {
-                        var acq = +params[1];
-                        this.automaticSession.acquisitionDate = +new Date(acq);
-                        this.automaticSession.acquisitionDate = this.automaticSession.acquisitionDate > 0 ? this.automaticSession.acquisitionDate : 0;
-                    }
-
-                    if (params.length > 2) {
-                        var renewal = +params[2];
-                        this.automaticSession.renewalDate = +new Date(renewal);
-                        this.automaticSession.renewalDate = this.automaticSession.renewalDate > 0 ? this.automaticSession.renewalDate : 0;
-                    }
-                } catch (e) {                    
-                    _InternalLogging.throwInternalNonUserActionable(LoggingSeverity.WARNING, "Error parsing ai_session cookie, session will be reset: " + Util.dump(e));
-                }            
-
-                if (this.automaticSession.renewalDate == 0) {
-                    _InternalLogging.throwInternalNonUserActionable(LoggingSeverity.WARNING, "AI sessoin renewal date is 0, session will be reset.");
+                this.initializeAutomaticSessionWithData(cookie.split("|"));
+            } else {
+                // There's no cookie, but we might have session data in local storage
+                // This can happen if the session expired or the user actively deleted the cookie
+                // We only want to recover data if the cookie is missing from expiry. We should respect the user's wishes if the cookie was deleted actively.
+                // The User class handles this for us and deletes our local storage object if the persistent user cookie was removed.
+                var userCookie = Util.getCookie('ai_user');
```

Comment:
This is a remnant from when I did checking for that cookie here. I'll remove that